scripts/azureml-assets/azureml/assets/validate_assets.py

```diff
@@ -1060,10 +1060,23 @@ def validate_assets(input_dirs: List[Path],
                 if check_environment_version:
                     error_count += validate_environment_version(asset_config)
 
-            if asset_config.type == assets.AssetType.PROMPT or asset_config.type == assets.AssetType.EVALUATIONRESULT:
-                error_count += validate_tags(asset_config, 'tag_values_shared.yaml')
+            if asset_config.type == assets.AssetType.EVALUATIONRESULT:
+                error_count += validate_tags(asset_config, 'evaluationresult/tag_values_shared.yaml')
+
+                asset_spec = asset_config._spec._yaml
+                evaluation_type = asset_spec.get('tags', {}).get('evaluation_type', None)
+
+                if evaluation_type == 'text_generation':
+                    valid_tags_filename = 'evaluationresult/tag_values_text_generation.yaml'
+                elif evaluation_type == 'text_embeddings':
+                    valid_tags_filename = 'evaluationresult/tag_values_text_embeddings.yaml'
+                else:
+                    raise ValidationException(f"Unknown evaluation type: {evaluation_type}")

```

Comment:
Could this validation check be moved to the validate spec section (which has a try/catch already) or could you include a try/catch here? Otherwise it won't reach the rest of the validation checks